### ISSUE TITLE
Warning for ignored files and ignored by default on linting

### ISSUE THREAD

**Issue by EvyiLdaRkmaN** [eslint/eslint#16602]
### Environment

Node version: v18.12.1
npm version: 8.19.2
Local ESLint version: 8.28.0
Global ESLint version: 8.28.0
Operating System: windows 11

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "extends": [
        "plugin:react/recommended",
        "airbnb",
        "prettier",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": "latest",
        "sourceType": "module",
        "project": ["./tsconfig.json"]
    },
    "plugins": [
        "react",
        "@typescript-eslint"
    ],
    "ignorePatterns": ["!.*"],
    "rules": {
       // ...
    }
}

```
</details>
<details>
<summary>eslintignore</summary>

```
node_modules
build
public
src/assets

#files
*.json
*.md
*.css
*.config.*
*.html
.*
```
</details>

## too many warnings when committing using husky
`package.json`
```json
"lint-staged": {
    "**/*": ["prettier --write --ignore-unknown", "eslint"]
  },
```

```js
// vite template code, using: npm create vite@latest pce-front -- --template react-ts
```

### What did you expect to happen?

I only expected the errors, since it is supposed that to avoid the mentioned warnings, I followed their recommendations and more

### What actually happened?

```
C:\Users\ameri\Documents\GitHub\pce-front\.eslintignore
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\.eslintrc.json
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\.gitignore
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\.husky\pre-commit
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\.prettierignore
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\.prettierrc.json
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override

C:\Users\ameri\Documents\GitHub\pce-front\index.html
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\package-lock.json
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\package.json
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\public\vite.svg
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\src\App.css
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\src\assets\react.svg
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\src\index.css
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\tsconfig.json
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\tsconfig.node.json
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

C:\Users\ameri\Documents\GitHub\pce-front\vite.config.ts
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @EvyiLdaRkmaN, thanks for the issue!

Can you please clarify what actually happened and what did you expect to happen? In particular:

* Was the commit successful?
* Do you expect not to see any warnings when the commit is successful, or do you expect to see warnings when there are any but not those about ignored files?

**Comment by EvyiLdaRkmaN** [comment]
Hi @mdjermanovic. 
I didn't expect so many warnings from files I'm specifically ignoring or even from config files, like .eslintignore, .eslintrc.json, .gitignore, and more.

>     * Was the commit successful?

yes, commit successful. Even though I don't know if it's right to ignore so many warnings

**Comment by nzakas** [comment]
There are two problems here:

1. You are explicitly ignoring those files in eslintignore, which is what’s causing the warnings. You should remove everything under “#files” in eslintignore. 
2. You should only run eslint with lint-staged on *.js files. Right now you are running on every file no matter the extension.

If you make those two changes the earnings will go away.

**Comment by EvyiLdaRkmaN** [comment]
Ok @nzakas .
therefore the following configuration would be correct?

.eslintignore

```
node_modules
build
public

#files
*.config.*
```

.package.json

```json
"lint-staged": {
    "**/*": "prettier --write --ignore-unknown",
    "*.{ts,tsx}": "eslint"
  },
```

This seems to no longer give warnings

**Comment by nzakas** [comment]
Yup, that's exactly it. 👍
